File: app/src/main/java/br/com/nicetuber/feature/channel_details/ChannelDetailsActivity.java
package br.com.nicetuber.feature.channel_details;

import android.support.v7.app.AlertDialog;

import org.parceler.Parcels;

import br.com.nicetuber.R;
import br.com.nicetuber.base.BaseActivity;
import br.com.nicetuber.databinding.ActivityChannelDetailsBinding;
import br.com.nicetuber.model.ChannelSearched;
import br.com.nicetuber.model.ChannelStatistics;
import br.com.nicetuber.util.ParamKey;

/**
 * Created by thiagozg on 22/10/2017.
 */

public class ChannelDetailsActivity
        extends BaseActivity<ActivityChannelDetailsBinding, ChannelDetailsPresenter>
        implements IChannelDetails.View {

    @Override
    protected void onStart() {
        super.onStart();
        ChannelSearched channelSearched =
                Parcels.unwrap(getIntent().getParcelableExtra(ParamKey.KEY_CHANNEL));
        binding.setChannelSearched(channelSearched);
        presenter.getChannelStatistics(channelSearched.getSnippet().getChannelId());
    }

    @Override
    protected void injectDagger() {
        getAppComponent().inject(this);
    }

    @Override
    protected int getXmlLayout() {
        return R.layout.activity_channel_details;
    }

    @Override
    protected void attachView() {
        presenter.attachView(this);
    }

    @Override
    protected void detachView() {
        presenter.detachView();
    }

    @Override
    public void showChannelStatistics(ChannelStatistics channelStatistics) {
        binding.setChannelStatistics(channelStatistics);
    }

    @Override
    public void showErrorMessage(String message) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle(getString(R.string.title_error_dialog))
                .setMessage(message)
                .setCancelable(false)
                .setPositiveButton(getString(R.string.ok), (dialog, id) -> finish())
                .show();
    }

}

File: app/src/main/java/br/com/nicetuber/feature/channel_details/ChannelDetailsActivity.kt
package br.com.nicetuber.feature.channel_details

import android.support.v7.app.AlertDialog
import br.com.nicetuber.R
import br.com.nicetuber.base.BaseActivity
import br.com.nicetuber.model.ChannelSearched
import br.com.nicetuber.model.ChannelStatistics
import br.com.nicetuber.util.formatThousand
import br.com.nicetuber.util.loadCircleGlide
import kotlinx.android.synthetic.main.activity_channel_details.*

/**
 * Created by thiagozg on 22/10/2017.
 */

class ChannelDetailsActivity : BaseActivity<ChannelDetailsPresenter>(),
        IChannelDetails.View {

    override fun onStart() {
        super.onStart()
        val channelSearched = intent.getParcelableExtra<ChannelSearched>(KEY_CHANNEL_SELECTED)
        bindData(channelSearched)
        presenter.getChannelStatistics(channelSearched.snippet?.channelId!!)
    }

    private fun bindData(channelSearched: ChannelSearched) {
        iv_channel.loadCircleGlide(channelSearched.snippet?.thumbnails?.imageHigh?.url)
        tv_title.text = channelSearched.snippet?.title
        tv_description.text = channelSearched.snippet?.description
    }

    override fun injectDagger() {
        appComponent?.inject(this)
    }

    override fun getXmlLayout(): Int {
        return R.layout.activity_channel_details
    }

    override fun attachView() {
        presenter.attachView(this)
    }

    override fun detachView() {
        presenter.detachView()
    }

    override fun showChannelStatistics(channelStatistics: ChannelStatistics) {
        tv_videos.text = channelStatistics.videoCount.formatThousand()
        tv_subscriber.text = channelStatistics.subscriberCount.formatThousand()
        tv_views.text = channelStatistics.viewCount.formatThousand()
        tv_comments.text = channelStatistics.commentCount.formatThousand()
    }

    override fun showErrorMessage(message: String) {
        val builder = AlertDialog.Builder(this)
        builder.setTitle(getString(R.string.title_error_dialog))
                .setMessage(message)
                .setCancelable(false)
                .setPositiveButton(getString(R.string.ok)) { dialog, id -> finish() }
                .show()
    }

    companion object {
        const val KEY_CHANNEL_SELECTED = "keyChannelSelected"
    }

}
